variables.tf

```diff
@@ -147,11 +157,21 @@ variable "cluster_egress_type" {
 variable "aks_pod_cidr" {
   description = "The CIDR to use for pod IP addresses. This field can only be set when network_plugin is set to kubenet. Changing this forces a new resource to be created."
   default     = "10.244.0.0/16"
+  validation {
+    condition     = var.aks_pod_cidr != "" ? can(cidrnetmask(var.aks_pod_cidr)) : true
+    error_message = "ERROR: aks_pod_cidr - value must either be null or must be a valid CIDR."
+  }
+
 }
 
 variable "aks_service_cidr" {
   description = "The Network Range used by the Kubernetes service. Changing this forces a new resource to be created."
   default     = "10.0.0.0/16"
+  validation {
+    condition     = var.aks_service_cidr != null ? can(cidrnetmask(var.aks_service_cidr)) : false
```

Comment:
Same as comment above regarding the cidrnetmask check including the empty string check.

Reply:
same as above